### PR TITLE
feat(dashboard): changelog news widget

### PR DESCRIPTION
**Pull request by scopsy**
### What changed? Why was the change needed?
- Added a new stacked changelog feed
- Cached for 4 hours in memory
- Conditionally shows the trial card UI 
- Only show the card if free trial is not visible

https://github.com/user-attachments/assets/8ae4120b-5346-423c-929b-77663edb31a7

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: apps/dashboard/src/components/side-navigation/changelog-cards.tsx [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+  const track = useTelemetry();
+
+  const fetchChangelogs = async (): Promise<Changelog[]> => {
+    const cachedData = storage.getCachedData();
+    if (cachedData) {
+      return filterChangelogs(cachedData, storage.getDismissedChangelogs());
+    }
+
+    const response = await fetch(CONSTANTS.CHANGELOG_API_URL);
+    const rawData = await response.json();
```
Comment (djabarovgeorge) [inline apps/dashboard/src/components/side-navigation/changelog-cards.tsx]: do we have any interface of this data?

File: apps/dashboard/src/components/side-navigation/changelog-cards.tsx [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+          />
+        ))}
+      </div>
+    </div>
+  );
+}
+
+const storage = {
+  getDismissedChangelogs: (): string[] => {
+    const dismissed = localStorage.getItem(CONSTANTS.DISMISSED_STORAGE_KEY);
```
Comment (djabarovgeorge) [inline apps/dashboard/src/components/side-navigation/changelog-cards.tsx]: i think we started to use Clerk context for such cases maybe we can utilize it in here as well

File: apps/dashboard/src/components/side-navigation/changelog-cards.tsx [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+  const fetchChangelogs = async (): Promise<Changelog[]> => {
+    const cachedData = storage.getCachedData();
+    if (cachedData) {
+      return filterChangelogs(cachedData, storage.getDismissedChangelogs());
+    }
+
+    const response = await fetch(CONSTANTS.CHANGELOG_API_URL);
+    const rawData = await response.json();
+
+    storage.cacheData(rawData);
```
Comment (djabarovgeorge) [inline apps/dashboard/src/components/side-navigation/changelog-cards.tsx]: why do we need to cache data if we use react query here?

**Review by djabarovgeorge — APPROVED** [review]
looks amazing :)

File: apps/dashboard/src/components/side-navigation/changelog-cards.tsx (hunk 1 above)
Comment (scopsy) [reply apps/dashboard/src/components/side-navigation/changelog-cards.tsx]: I want to cache it between reloads, so we don't always make api calls to product lane on each page load as this info rarely changes. But, good point

**Pull request merged**